lib/cocoapods/installer/analyzer/target_generator.rb

```diff
@@ -0,0 +1,397 @@
+          end
+          pod_targets.each do |target|
+            all_specs = all_resolver_specs.group_by(&:name)
+            dependencies = dependencies_for_specs(target.non_test_specs.to_set, target.platform, all_specs.dup).group_by(&:root)
+            target.dependent_targets = filter_dependencies(dependencies, pod_targets_by_name, target)
+            target.test_dependent_targets_by_spec_name = target.test_specs.each_with_object({}) do |test_spec, hash|
+              test_dependencies = dependencies_for_specs([test_spec], target.platform, all_specs).group_by(&:root)
+              test_dependencies.delete_if { |k| dependencies.key? k }
+              hash[test_spec.name] = filter_dependencies(test_dependencies, pod_targets_by_name, target)
+            end
+          end
+        end
+
+        #-----------------------------------------------------------------------#
+
+        private
+
+        # @!group Target Generation Helpers
+
+        # Generate an aggregate target for a single user target
+        #
+        # @param  [TargetInspection] target_inspection
+        #         the target inspection for the target definition
+        #
+        # @param  [Hash{TargetDefinition => Array<TargetInspectionResult>}] target_inspections
+        #         the user target inspections used to construct the aggregate and pod targets.
+        #
+        # @param  [Array<PodTarget>] pod_targets
+        #         the pod targets, which were generated.
+        #
+        # @param  [Array<ResolvedSpecification>}] resolver_specs
+        #         the resolved for the target.
+        #
+        # @param  [PodfileDependencyCache] dependency_cache
+        #         the podfile dependency cache
+        #
+        # @param  [Sandbox] sandbox
+        #         the sandbox into which the targets will be installed
+        #
+        # @param  [InstallationOptions] installation_options
+        #         the installation_options to use while generating the target
+        #
+        # @return [AggregateTarget]
+        #
+        def self.generate_aggregate_target(target_definition, target_inspections, pod_targets, resolver_specs, dependency_cache, sandbox, installation_options)
+          target_requires_64_bit = requires_64_bit_archs?(target_definition.platform)
+          if installation_options.integrate_targets?
+            target_inspection = target_inspections[target_definition]
+            raise "missing inspection: #{target_definition.name}" unless target_inspection
+            user_project = target_inspection.project
+            client_root = user_project.path.dirname.realpath
+            user_target_uuids = target_inspection.project_target_uuids
+            user_build_configurations = target_inspection.build_configurations
+            archs = target_requires_64_bit ? ['$(ARCHS_STANDARD_64_BIT)'] : target_inspection.archs
+          else
+            user_project = nil
+            client_root = Config.instance.installation_root
```

Comment:
not happy about this part, but trying to keep the number of arguments from growing to a paragraph